src/main/lpfConverter.ts

```diff
@@ -0,0 +1,304 @@
+// ==LICENSE-BEGIN==
+// Copyright 2017 European Digital Reading Lab. All rights reserved.
+// Licensed to the Readium Foundation under one or more contributor license agreements.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file exposed on Github (readium) in the project repository.
+// ==LICENSE-END==
+
+import * as debug_ from "debug";
+import { app } from "electron";
+import { copy, ensureDir, move } from "fs-extra";
+import * as moment from "moment";
+import { basename, extname, join } from "path";
+import { TaJsonSerialize } from "r2-lcp-js/dist/es6-es2015/src/serializable";
+import { Metadata } from "r2-shared-js/dist/es6-es2015/src/models/metadata";
+import { Link } from "r2-shared-js/dist/es6-es2015/src/models/publication-link";
+import { streamToBufferPromise } from "r2-utils-js/dist/es6-es2015/src/_utils/stream/BufferUtils";
+import { IStreamAndLength } from "r2-utils-js/dist/es6-es2015/src/_utils/zip/zip";
+import { injectBufferInZip } from "r2-utils-js/dist/es6-es2015/src/_utils/zip/zipInjector";
+import { acceptedExtensionObject } from "readium-desktop/common/extension";
+import { JsonMap } from "readium-desktop/typings/json";
+import { ObjectKeys } from "readium-desktop/utils/object-keys-values";
+import { v4 as uuidV4 } from "uuid";
+
+import { Publication as R2Publication } from "@r2-shared-js/models/publication";
+import { zipLoadPromise } from "@r2-utils-js/_utils/zip/zipFactory";
+import { iso8601DurationsToSeconds } from "readium-desktop/utils/iso8601";
+
+// Logger
+const debug = debug_("readium-desktop:main#lpfConverter");
+
+async function copyAndRenameLpfFile(lpfPath: string): Promise<string> {
+
+    const userPath = app.getPath("userData");
```

Comment:
I would suggest using a proper temporary directory instead.
`import { tmpNameSync } from "tmp";`
```
const tmpFolderPath = tmpNameSync({
            dir: "LPF",
            prefix: "readium-desktop-",
            postfix: ".extension"});
```

Reply:
`tmpNameSync` is the synchronous version of tmpName and should be `tmpName` instead

https://github.com/benjamingr/tmp-promise#synchronous-filename-generation
https://github.com/benjamingr/tmp-promise#asynchronous-filename-generation

this library store the file in the OS tmp directory: It's used the `os.tmpdir()` method to this.
https://github.com/raszi/node-tmp/blob/61f9a7ddda5d183ca790a214ee5981ef470ddb33/lib/tmp.js#L578

i just switch `app.getPath("userData");` to `os.tmpdir()` and should be the same.